### PR TITLE
nrf5x: Add UART modecfg feature implementation

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

This PR adds UART modecfg feature support for the nrf5x devices

### Testing procedure

validate with tests/periph_uart_modecfg

### Issues/PRs references

None

### CODE REVIEW

File: cpu/nrf5x_common/periph/uart.c [hunk 1]
```diff
@@ -275,6 +275,43 @@ void uart_poweroff(uart_t uart)
...
+    if (data_bits != UART_DATA_BITS_8) {
+        return UART_NOMODE;
+    }
+
+    if (parity != UART_PARITY_NONE && parity != UART_PARITY_EVEN) {
+        return UART_NOMODE;
+    }
+
+    uart_poweroff(uart);
+    dev(uart)->ENABLE = 0;
```
Comment (bergzand) [inline cpu/nrf5x_common/periph/uart.c]: Not sure if it is required to disable the controller before changing this. I put it in originally when things didn't work:tm: as expected. It is probably not required anymore (same with the enable below)

Comment (bergzand) [reply cpu/nrf5x_common/periph/uart.c]: Tested, powering down before and powering up after changing the mode is not required. Removed the code.

File: cpu/nrf5x_common/periph/uart.c [hunk 2]
```diff
@@ -275,6 +275,39 @@ void uart_poweroff(uart_t uart)
...
+
+    if (stop_bits == UART_STOP_BITS_2) {
+        dev(uart)->CONFIG |= UARTE_CONFIG_STOP_Msk;
+    }
+    else {
+        dev(uart)->CONFIG &= ~UARTE_CONFIG_STOP_Msk;
+    }
+
+    if (parity == UART_PARITY_EVEN) {
+        dev(uart)->CONFIG |= UART_CONFIG_PARITY_Msk;
```
Comment (aabadie) [inline cpu/nrf5x_common/periph/uart.c]: Should be UART**E**_CONFIG_PARITY_Msk. Same below.

Comment (bergzand) [reply cpu/nrf5x_common/periph/uart.c]: Fixed!

**Comment by bergzand** [comment]
squashed!

**Comment by aabadie** [comment]
and go!

**Pull request merged**